### DISCUSSION TITLE
Carico magazzino tramite Acquisto

### DISCUSSION THREAD

**Discussion by marioquark** [fattureincloud/api#414] [Q&A] [answered]
Ciao. Vorrei eseguire un carico di magazzino su prodotto specifico.
Leggo che l'API per carico "diretto" non è stata esposta e che quindi devo operare facendo un inserimento di un Acquisto, che risulterà anche in un carico di prodotto.
Sto quindi usando l'API **"Create Received Document" corredata di "items_list" e "code" con "qty"** del prodotto da caricare, ma non sta funzionando: **inserisce l'acquisto, ma non vedo carico di stock**.
Purtroppo dalla documentazione non è chiaro come operare per il mio scopo, la spiegazione dei campi è molto generica.
Sto lavorando con questo JSON. Cosa devo modificare? Grazie!
`{
   "data":{
      "date":"2024-02-16",
      "amount_net":0,
      "description":"MIA DESCRIZIONE",
      "amortization":1,
      "type":"expense",
      "payments_list":[
         {
            "amount":0,
            "due_date":"2024-02-16",
            "payment_terms":{
               "type":"standard",
               "days":0
            },
            "id":0,
            "paid_date":"2024-02-16",
            "status":"paid"
         }
      ],
      "is_detailed":false,
      "next_due_date":"2024-02-16",
      "amount_withholding_tax":0,
      "rc_center":"",
      "is_marked":false,
      "amount_vat":0,
      "e_invoice":false,
      "items_list":[
         {
            "code":"AAA",
            "vat":{
               "description":"",
               "id":0,
               "value":22
            },
            "not_taxable":false,
            "qty":1234,
            "name":"MIA DESCRIZIONE",
            "gross_price":0,
            "stock":true
         }
      ],
      "amount_other_withholding_tax":0,
      "invoice_number":"",
      "amount_gross":0,
      "entity":{
         "name":"Mia Azienda Srl"
      }
   }
}`

**Comment by emajo** [comment]
Ciao @marioquark il problema sta nel campo `is_detailed`, settato a false fa sì che vengano ignorati i campi `items_list` e `payments_list`.

Grazie,
Emanuele.

**Comment by marioquark** [comment]
Grazie @emajo per la pronta risposta.
Ho ritentato con il flag _is_detailed_ a True, ma continua a non funzionare: non vedo movimenti di magazzino.

Sto immettendo il prodotto settando _code_ e _description_ e non dichiarando _product_id_ (che non conosco nel sistema che invia).

E' strano perchè nel json di ritorno mi dice
```
            "product_id":null,
            "code":"AAA",
            "name":"PRODOTTO AAA",
```
AAA è un prodotto che ho in anagrafica.

Ho provato anche inserendo solo code, non modifica il magazzino.

Grazie ancora

**Comment by emajo** [comment, marked as answer]
Si @marioquark, ho dimenticato di specificare che la movimentazione del magazzino viene effettuata solamente settando il campo `product_id`, non tramite `code`, se serve recuperare l'id del prodotto partendo dal code è necessario fare una chiamata di list dei prodotti impostando il filtro sul code, in questo modo: `GET https://api-v2.fattureincloud.it/c/{company_id}/products?q=code = 'codice_prodotto'`.

Grazie,
Emanuele.

**Comment by marioquark** [comment]
Grazie @emajo ora riesco a inserire i carichi di magazzino!
Un ultimo dettaglio: come posso personalizzare nella pagina Magazino la voce Descrizione?

Io inserisco 
```
"data": {
"type": "expense",
"description": descrizione,
...
```
ma la descrizione viene fuori solo nella pagina Acquisti, mentre sotto magazzino leggo un generico "Aquisto - [nome formitore]"

Grazie anora

**Reply by emajo** [reply]
Ciao @marioquark, scusami per la risposta tardiva ma mi ero perso la notifica, per quanto riguarda la descrizione del movimento è personalizzabile solo sui movimenti manuali, non quelli generati da fatture o aquisti.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Buongiorno @marioquark 
abbiamo risposto alla sua domanda?

Grazie
Mauro

**Reply by marioquark** [reply]
Si, se non si può fare la customizzazione della dicitura ci accontentiamo di quello che è fattibile ad oggi.
Grazie molte
